persistence-api/src/main/java/io/stargate/db/PagingPosition.java

```diff
@@ -59,6 +62,44 @@
         .collect(Collectors.toMap(e -> e.getKey().name(), Entry::getValue));
   }
 
+  @Value.Lazy
+  default TableName table() {
+    Set<TableName> tables =
+        currentRow().keySet().stream()
+            .map(
+                c ->
+                    ImmutableTableName.builder()
+                        .keyspace(Objects.requireNonNull(c.keyspace()))
+                        .name(Objects.requireNonNull(c.table()))
+                        .build())
+            .collect(Collectors.toSet());
+
+    if (tables.isEmpty()) {
+      throw new IllegalArgumentException("Missing table information in custom paging request.");
+    }
+
+    if (tables.size() > 1) {
+      throw new IllegalArgumentException("Too many tables are referenced: " + tables);
+    }
+
+    return tables.iterator().next();
+  }
+
+  default ByteBuffer requiredValue(String columnName) {
+    TableName table = table(); // indirectly validates that there's only one table referenced
+
+    ByteBuffer value = currentRowValuesByColumnName().get(columnName);
```

Comment:
Interesting implementation detail. Lazy initialized values in the generated immutable wrapper are implemented  in a thread-safe way (`lazyInitBitmap` is volatile). There's probably a primary key size threshold where a linear search without caching the map is much faster. Maybe document store is bigger than this threshold with its big-ish primary key?
```java
  public Map<String, ByteBuffer> currentRowValuesByColumnName() {
    if ((lazyInitBitmap & CURRENT_ROW_VALUES_BY_COLUMN_NAME_LAZY_INIT_BIT) == 0) {
      synchronized (this) {
        if ((lazyInitBitmap & CURRENT_ROW_VALUES_BY_COLUMN_NAME_LAZY_INIT_BIT) == 0) {
          this.currentRowValuesByColumnName = Objects.requireNonNull(PagingPosition.super.currentRowValuesByColumnName(), "currentRowValuesByColumnName");
          lazyInitBitmap |= CURRENT_ROW_VALUES_BY_COLUMN_NAME_LAZY_INIT_BIT;
        }
      }
    }
    return currentRowValuesByColumnName;
  }
```

Reply:
I agree, there are ways to improve this code, but I did not want to do any "preliminary optimization" :) At this point Docs API has other areas where the efficiency can be improved with more gain, I believe.

Also, for there should be at most one custom pagination requests per Docs API HTTP call, so hash lookup inefficiencies may not even manifest :shrug:

Reply:
Is it possible to micro-benchmark this (i..e using JMH)?

Reply:
Certainly, but I'm just not sure it's worth the time at this point. I assume we'll test it out as part of our Docs API tests and if it shows up as a bottleneck we'll optimize. Would that be ok?

Reply:
At this point Docs API is the only "client" of `PagingPosition`... other REST endpoints may need it too, but only when and if we support more advanced queries.

Reply:
I'm fine with merging it as is.